deepchecks/vision/datasets/detection/mask.py

```diff
@@ -0,0 +1,292 @@
+from torchvision.datasets.utils import download_and_extract_archive
+from torchvision.transforms import transforms
+from typing_extensions import Literal
+
+from deepchecks import vision
+from deepchecks.vision import DetectionData
+from deepchecks.vision.vision_data import IndicesSequentialSampler
+
+__all__ = ['load_dataset', 'load_model', 'MaskData', 'MaskDataset']
+
+
+DATA_DIR = Path(__file__).absolute().parent
+
+
+class MaskPrecalculatedModel(nn.Module):
+    """Model that returns pre-calculated predictions for the mask detection dataset."""
+
+    def __init__(self, device: t.Union[str, torch.device] = 'cpu'):
+        super().__init__()
+        self._pred_dict_url = 'https://ndownloader.figshare.com/files/38116641'
+        pred_dict_path = os.path.join(DATA_DIR, 'pred_dict.json')
+        urllib.request.urlretrieve(self._pred_dict_url, pred_dict_path)
+        with open(pred_dict_path, 'r', encoding='utf8') as f:
+            self._pred_dict = json.load(f)
+
+        self._device = device
+
+    def forward(self, batch):
+        image_hashes = [self._hash_image((img.cpu().numpy() if isinstance(img, torch.Tensor) else img))
+                        for img in batch[0]]
+        return [torch.tensor(self._pred_dict[image_hash]).to(self._device) for image_hash in image_hashes]
+
+    @staticmethod
+    def _hash_image(img):
+        return hashlib.sha1(img).hexdigest()
+
+
+def load_model(device: t.Union[str, torch.device] = 'cpu') -> nn.Module:
+    """Load the pre-calculated prediction model and return it."""
+    dev = torch.device(device) if isinstance(device, str) else device
+
+    return MaskPrecalculatedModel(device=dev)
+
+
+class MaskData(DetectionData):
+    """Class for loading the mask dataset, inherits from :class:`~deepchecks.vision.DetectionData`.
+
+    Implement the necessary methods to load the dataset.
+    """
+
+    def batch_to_labels(self, batch) -> List[torch.Tensor]:
+        """Convert the batch to a list of labels."""
+
+        def extract_dict(in_dict):
+            return torch.concat([in_dict['labels'].reshape((-1, 1)), in_dict['boxes']], axis=1)
+
+        return [extract_dict(tensor) for tensor in batch[1]]
+
+    def batch_to_images(self, batch) -> Iterable[np.ndarray]:
+        """Convert the batch to a list of images."""
+        return [np.array(x.permute(1, 2, 0)) * 255 for x in batch[0]]
+
+    def infer_on_batch(self, batch, model, device: t.Union[str, torch.device] = 'cpu') -> List[torch.Tensor]:
+        """Infer on a batch using the given model."""
+        return model(batch)
+
+
+def _batch_collate(batch):
+    return tuple(zip(*batch))
+
+
+def load_dataset(
+        time_index: int = 0,
+        batch_size: int = 32,
+        num_workers: int = 0,
+        shuffle: bool = True,
+        pin_memory: bool = True,
+        object_type: Literal['VisionData', 'DataLoader'] = 'DataLoader'
+) -> t.Union[DataLoader, vision.VisionData]:
+    """Get the mask dataset and return a dataloader.
```

Comment:
Move the description of the dataset to top of the page
I think this way docstring compile better

Reply:
Can do